src/room.ts

```diff
@@ -171,23 +173,89 @@ export class Room extends EventEmitter implements Sayable {
+   * This is Sayable for all listeners.
+   * Which means there will be a this.say() the method inside listener call,
+   * you can use it sending a message to filehelper, just for logging/reporting usage for your convenience.
+   * @param {'leave'} event
+   * @param {(this: Room, leaver: Contact) => void} listener
+   * @returns {this}
+   * @example
+   * ```ts
+   *  const room = await Room.find({topic: 'event-room'}) // change `event-room` to any room topic in your wechat
+   *  if (room) {
+   *    room.on('leave', (room: Room, leaverList: Contact[]) => {
+   *      const nameList = leaverList.map(c => c.name()).join(',')
+   *      console.log(`Room ${room.topic()} lost member ${nameList}`)
+   *    })
+   *  }
+   * ```
    */
   public on(event: 'leave', listener: (this: Room, leaver: Contact) => void): this
+
   /**
-   * @todo document me
+   * Get room join event, emitted when someone join the room
+   * This is Sayable for all listeners.
+   * Which means there will be a this.say() the method inside listener call,
+   * you can use it sending a message to filehelper, just for logging/reporting usage for your convenience.
+   *
+   * @param {'join'} event
+   * @param {(this: Room, inviteeList: Contact[] , inviter: Contact)  => void} listener
+   * @returns {this}
+   * @example
+   * ```ts
+   *  const room = await Room.find({topic: 'event-room'}) // change `event-room` to any room topic in your wechat
+   *  if (room) {
+   *    room.on('join', (room: Room, inviteeList: Contact[], inviter: Contact) => {
+   *      const nameList = inviteeList.map(c => c.name()).join(',')
+   *      console.log(`Room ${room.topic()} got new member ${nameList}, invited by ${inviter}`)
+   *    })
+   *  }
    */
   public on(event: 'join' , listener: (this: Room, inviteeList: Contact[] , inviter: Contact)  => void): this
+
   /**
-   * @todo document me
+   * Get topic event, emitted when someone change room topic
+   * This is Sayable for all listeners.
+   * Which means there will be a this.say() the method inside listener call,
+   * you can use it sending a message to filehelper, just for logging/reporting usage for your convenience.
+   *
+   * @param {'topic'} event
+   * @param {(this: Room, topic: string, oldTopic: string, changer: Contact) => void} listener
+   * @returns {this}
+   * @example
+   * ```ts
+   *  const room = await Room.find({topic: 'event-room'}) // change `event-room` to any room topic in your wechat
+   *  if (room) {
+   *    room.on('topic', (room: Room, topic: string, oldTopic: string, changer: Contact) => {
+   *      console.log(`Room ${room.topic()} topic changed from ${oldTopic} to ${topic} by ${changer.name()}`)
+   *    })
+   *  }
    */
   public on(event: 'topic', listener: (this: Room, topic: string, oldTopic: string, changer: Contact) => void): this
+
   /**
-   * @todo document me
+   * Get EVENT_PARAM_ERROR event
```

Comment:
The `EVENT_PARAM_ERROR` is just a guard for IDE Intelligence.

Need not be documented, and should be set to `@private`